text/0000-template.md

```diff
@@ -0,0 +1,90 @@
+
+## Motivation
+
+The difference between `{{let}}` and `{{with}}` is with how they handle conditional arguments. The `{{let}}` helper's block content is always rendered, regardless of its parameters. In contrast, `{{with}}` only renders its main block when the first position parameter is truthy. For example:
+
+```hbs
+{{#with "Alex" as |value|}}
+  {{value}} is truthy
+{{else}}
+  The first positional param was falsy
+{{/with}}
+```
+
+Will render "[Alex] is truthy".
+
+```hbs
+{{#with false as |value|}}
+  {{value}} is truthy
+{{else}}
+  The first positional param was falsy
+{{/with}}
+```
+
+Will render "The first positional param was falsy".
+
+As the naming of `{{with}}` doesn't give any indication of its behavior, we should deprecate it and introduce the better named `{{if-let}}` helper. Doing do will make Ember easier to learn and use.
+
+Here's how the `{{with}}` examples above would look like when using `{{if-let}}`:
+
+```hbs
+{{#if-let "Alex" as |value|}}
+  {{value}} is truthy
+{{else}}
+  The first positional param was falsy
+{{/if-let}}
+```
+
+Will render "[Alex] is truthy".
+
+```hbs
+{{#if-let false as |value|}}
+  {{value}} is truthy
+{{else}}
+  The first positional param was falsy
+{{/if-let}}
+```
+
+## Detailed design
+
+We'll create a new `{{if-let}}` helper which will have the same behavior to the current [`{{with}}` helper](https://github.com/glimmerjs/glimmer-vm/blob/a15a7c86ae50a882f3f395a394cdaff6211ade30/packages/%40glimmer/opcode-compiler/lib/syntax/builtins.ts#L81-L106).
+
+We'll create an AST transform in [`packages/ember-template-compiler`](https://github.com/emberjs/ember.js/tree/master/packages/ember-template-compiler) which will emit a deprecation warning for all uses of `{{with}}`. The deprecation warning will be something like:
+
+```
+Using `{{with}}` is deprecated, please use `{{let}}` or `{{if-let}}` instead.
+```
+
+This new deprecation warning should link to some documentation which clearly describes when you should use `{{let}}` and when you should use `{{let-if}}`.
+
+## How we teach this
+
+We'll add `{{if-let}}` documentation to the [`Ember.Templates.helpers` API docs](https://www.emberjs.com/api/ember/3.7/classes/Ember.Templates.helpers). The new documentation should make the unique use cases of `{{if-let}}` clear. 
+
+We'll mentiton the new `{{if-let}}` helper in an Ember point release blog post.
+
+The deprecation message will link to good documentation which describe why `{{with}}` was deprecated and gives clear guidelines on how to migrate to using either `{{let}}` or `{{if-let}}`.
+
+## Drawbacks
+
+This adds a litte API and documentation churn.
+
+## Alternatives
```

Comment:
There's probably a very good reason for not doing it but just throwing this out there. Rather than increase the template API surface area could we enhance `{{let}}` with the `{{else}}` functionality?

Reply:
@patocallaghan see the [motivation section](https://github.com/emberjs/rfcs/blob/master/text/0286-block-let-template-helper.md#motivation) from the `{{let}}` RFC:

> While the conditional semantics of with are coherent with the other built-in helpers like each and if, users often find this unexpected. The fact that only the first positional parameter of with controls whether the block is rendered might also add to the confusion.

`{{let}}` was purposefully designed to always render its block.